### PR TITLE
Update Release Process to 3.9.0 changes

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**
Updates the release process file for 3.9.0 progress.

**Context: Fixes an issue? Related issues**
No. However: Related to the ARM changes

**Status of this Pull Request**
Ready to merge

**What is missing until this pull request can be merged?**
Nothing. Only one review

**Does this need translation?**

NO.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I'm sure that this Pull Request goes to the correct branch

### CODE REVIEW

File: contribute/en/Release-Process.md [hunk 1]
```diff
@@ -248,24 +248,24 @@ Current state: <!-- Planning|Translations (beta)|Code freeze (rc)|Released -->
...
 - [ ] Pin this issue
-- [ ] Ensure that all issues/PR targeted for this release are done by checking the Project board with [the appropriate filter for this release](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.2%22). Remind main developers to review entries in *Waiting on team* state. 
+- [ ] Ensure that all issues/PR targeted for this release are done by checking the Project board with [the appropriate filter for this release](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.2%22). Remind main developers to review entries in *Waiting on team* state.
 - [ ] Agree to de-tag unfinished Issues/PRs.
 - [ ] Declare a freeze for code and website by updating this Issue and adding a comment. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
 - [ ] Check the `needs documentation` label for any outstanding PRs flagged for this release and remove that label if done.
 - [ ] Check ./Jamulus -h output against the Include-Client/Server-Commands.md pages (wiki/en) and man page (distributions/Jamulus.1). Update if necessary.
```
Comment (pljones) [inline contribute/en/Release-Process.md]: Link to github location would be useful.

File: contribute/en/Release-Process.md [hunk 2]
```diff
@@ -328,7 +328,8 @@ Please find all the details in the [Changelog](https://github.com/jamulussoftwar
 _Windows users: Please note that in the first days after release SmartScreen will probably display warnings about this release being unknown upon download and/or execution of the installer. Let us know when you do not see this warning anymore and we will update this announcement accordingly._
 
 **[↓ Windows](<!-- direct link to Windows version -->)** (ASIO version), alternative: [↓ JACK version](<!-- direct link to JACK version -->)
-**[↓ macOS](<!-- direct link to macOS SIGNED version -->)** for High Sierra (10.13) to Big Sur (11) and [↓ macOS legacy build](<!-- direct link to macOS legacy version -->) (unsigned) for macOS Sierra (10.12), El Capitan (10.11) or Yosemite (10.10).
+**[↓ macOS (Intel)](<!-- direct link to macOS SIGNED Intel version -->)** for Catalina (10.15) and higher and [↓ macOS legacy build](<!-- direct link to macOS legacy version -->) (unsigned) for macOS Mojave (10.14) down to Yosemite (10.10).
+**[↓ macOS (Apple Silicon)](<!-- direct link to macOS SIGNED ARM version -->)** for Big Sur (11.0) or later (needs ARM chip)
```
Comment (pljones) [inline contribute/en/Release-Process.md:332]: How do Apple refer to these?  M-series?

Comment (ann0see) [reply contribute/en/Release-Process.md:332]: Not sure. I think it's apple silicon
https://support.apple.com/en-us/HT211814 or Apple Chip

File: contribute/en/Release-Process.md (hunk 1 above)
Comment (ann0see) [reply contribute/en/Release-Process.md]:
```suggestion
- [ ] Check ./Jamulus -h output against the [Include-Client](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Client-Commands.md)/[Server-Commands](https://github.com/jamulussoftware/jamuluswebsite/blob/next-release/wiki/en/Include-Server-Commands.md).md pages and [man page (distributions/Jamulus.1)[(https://github.com/jamulussoftware/jamulus/blob/master/distributions/Jamulus.1). Update if necessary.
```

File: contribute/en/Release-Process.md (hunk 2 above)
Comment (pljones) [reply contribute/en/Release-Process.md:332]: Yep seems to be okay.

**Pull request merged**